tests/src/test/scala/akka/kafka/scaladsl/IntegrationSpec.scala

```diff
@@ -189,7 +189,7 @@ class IntegrationSpec extends SpecBase(kafkaPort = KafkaPorts.IntegrationSpec) w
       }
 
       control2 should not be null
-      sleep(4.seconds)
+      sleep(10.seconds)
```

Comment:
This test-case is timing dependent. Try increasing this even more to see if the test-case starts to pass when running all of the test-cases in the spec.

Ideally we should refactor the test to not be timing dependent.

Reply:
Is there a reason for the initial value of four or it was just  derived by trial and error.

Reply:
Yup, trial and error. 

The ideal solution would be to not wait for a defined amount of time, but instead look at the arriving elements and complete the consumer stream when some condition is met. Take a look here for inspiration:

https://github.com/akka/alpakka-kafka/blob/9a08f599bd3476ffcd54b574933a433b8058c2c9/tests/src/test/scala/akka/kafka/scaladsl/PartitionedSourcesSpec.scala#L361

But we can leave this improvement for another PR.

Reply:
We can indeed but still these random failures are quite strange. I slashed the number of events by 4 and increased the sleep duration and am still getting failures. I cannot quite explain it to myself why my change has triggered this strange behaviour...

Reply:
Aha. If decreasing the number of messages and increasing sleep times does not solve this, then maybe indeed we are loosing messages now. I'll dig deeper as well.

Reply:
Its strange indeed. I took another look and only changing the poll call to the deprecated once makes all the tests pass. Looking at the semantics of the new poll call, the main difference is the fact that it might block for infinity on fetching metadata... Will have to think a bit more on that one.